docs/src/stake-pool.md

```diff
@@ -1,11 +1,427 @@
+Save this seed phrase to recover your new keypair:
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+============================================================================
+$ solana create-stake-account stake-account.json 10
+Signature: 5Y9r6MNoqJzVX8TWryAJbdp8i2DvintfxbYWoY6VcLEPgphK2tdydhtJTd3o3dF7QdM2Pg8sBFDZuyNcMag3nPvj
+$ solana delegate-stake 4F4AYKZbNtDnu7uQey2Vkz9VgkVtLE6XWLezYjc9yxZa 2HUKQz7W2nXZSwrdX5RkfS2rLU4j1QZLjdGCHcoUKFh3
+Signature: 2cDjHXSHjuadGQf1NQpPi43A8R19aCifsY16yTcictKPHcSAXN5TvXZ58nDJwkYs12tuZfTh5WVgAMSvptfrKdPP
+```
+
+Two epochs later, when the stake is fully active and has received one epoch of
+rewards, we can deposit the stake into the stake pool.
+
+```sh
+$ spl-stake-pool deposit --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC --stake 4F4AYKZbNtDnu7uQey2Vkz9VgkVtLE6XWLezYjc9yxZa
+Depositing into stake account FYQB64aEzSmECvnG8RVvdAXBxRnzrLvcA3R22aGH2hUN
+Creating account to receive tokens 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
+Signature: 4AESGZzqBVfj5xQnMiPWAwzJnAtQDRFK1Ha6jqKKTs46Zm5fw3LqgU1mRAT6CKTywVfFMHZCLm1hcQNScSMwVvjQ
+```
+
+In return, the stake pool has sent us staking derivatives in the form of SPL
+tokens.  We can double-check our stake pool account using the SPL token
+command-line utility.
+
+```sh
+$ spl-token balance 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
+0.024058966
+```
+
+### Example: Update
+
+Every epoch, the network pays out rewards to stake accounts managed by the stake
+pool, increasing the value of staking derivative SPL tokens minted on deposit.
+In order to calculate the proper value of these stake pool tokens, we must update
+the total value managed by the stake pool every epoch.
+
+The Solana transaction processor has two important limitations:
+
+* size of the overall transaction, limited to roughly 1 MTU / packet
+* computation budget per instruction
+
+A stake pool may manage hundreds of staking accounts, so it is impossible to
+update the total value of the stake pool in one instruction. Thankfully, the
+command-line utility does all of the work of breaking up transactions.
+
+```sh
+$ spl-stake-pool update --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC 
+Signature: 3Yx1RH3Afqj5ckX8YvPCRt1DudVP4HuRPkh1dBPvTM9GqGxcB9ZXHGZPADVSZiaqKi166fevMG232EWxrRWswPtt
+```
+
+If another user already updated the stake pool balance for the current epoch, we
+see a different output.
+
+```sh
+$ spl-stake-pool update --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC 
+Stake pool balances are up to date, no update required.
+```
+
+### Example: Withdraw stake
+
+Whenever we want to recover SOL plus accrued rewards, we can provide our
+staking derivative SPL tokens in exchange for an activated stake account.
+
+Let's withdraw 0.02 of our staking derivative tokens from the stake pool.
+
+```sh
+$ spl-stake-pool withdraw --pool 3CLwo9CntMi4D1enHEFBe3pRJQzGJBCAYe66xFuEbmhC  --amount 0.02 --burn-from 34XMHa3JUPv46ftU4dGHvemZ9oKVjnciRePYMcX3rjEF
```

Comment:
The `--burn-from` argument name feels unsettling.  This is "my" stake pool SPL Token account right?  `--withdraw-from`?

> Creating account to receive stake CZF2z3JJoDmJRcVjtsrz1BKUUGNL3VPW5FPFqge1bzmQ

Can the recipient stake address be passed in?  Having to parse the cli output to determine the address will be hard to use from a script

Reply:
Will change the param.

And the stake address can be passed in, but it's bugged currently since it always splits.  Added #1421 for this